R/vars-pull.R

```diff
@@ -32,17 +32,21 @@
 #' # You can unquote variables:
 #' var <- 10
 #' vars_pull(letters, !!var)
-vars_pull <- function(vars, var = -1, error_call = caller_env()) {
+vars_pull <- function(vars, var = -1, error_call = caller_env(), error_arg = "var") {
   expr <- enquo(var)
+  if (quo_is_missing(expr)) {
+    expr <- -1
+  }
+
```

Comment:
Would this give better results?


```r
error_arg <- error_arg %||% as_label(expr)
```

Reply:
If not, `error_arg = ""` is probably better. This should be a value supported by vctrs.

Reply:
The first option doesn't look good, and the second option requires more special casing of the manual abort in `pull_as_location2()`. Given that `pull()` is the primary use of this function, and it's argument is called `var`, I think it's a reasonable tradeoff. We can do better in `eval_pull()`.